### PR TITLE
Commerce 1539 (4/7) Clean unused code / logic and implement missing methods

### PR DESCRIPTION
**Pull request by alecsloan**

### CODE REVIEW

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
@marco-leo Rebased

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

File: commerce-product-type-virtual-order-service/src/main/java/com/liferay/commerce/product/type/virtual/order/service/http/CommerceVirtualOrderItemServiceHttp.java [hunk 1]
```diff
@@ -69,12 +69,7 @@
 				returnObj = TunnelUtil.invoke(httpPrincipal, methodHandler);
 			}
 			catch (Exception e) {
-				if (e instanceof Exception) {
-					throw (Exception)e;
-				}
-
-				throw new com.liferay.portal.kernel.exception.SystemException(
-					e);
+				throw e;
```
Comment (ethanbustad) [inline commerce-product-type-virtual-order-service/src/main/java/com/liferay/commerce/product/type/virtual/order/service/http/CommerceVirtualOrderItemServiceHttp.java]: doesn't this mean we can just remove the `catch` block?

**Review by ethanbustad — CHANGES_REQUESTED** [review]
Can we talk about the `Implement equals() and hashCode() to prevent potential bugs` commit tomorrow? From what I can see, the superclasses of most of these already define `equals` and `hashCode`, and in pretty much the exact same way, so I'm not sure why this is necessary....

**Comment by alecsloan** [comment]
@ethanbustad soudns good, from what I remember I thought it was weird too that it was complaining about `equals() and hashcode()` but I do recall it telling me they needed to be overridden.

File: commerce-product-type-virtual-order-service/src/main/java/com/liferay/commerce/product/type/virtual/order/service/http/CommerceVirtualOrderItemServiceHttp.java (hunk 1 above)
Comment (alecsloan) [reply commerce-product-type-virtual-order-service/src/main/java/com/liferay/commerce/product/type/virtual/order/service/http/CommerceVirtualOrderItemServiceHttp.java]: Yes, and that we don't need the returnObj variable either

**Comment by alecsloan** [comment]
ci:test

**Comment by ethanbustad** [comment]
Just talked this over in person -- Alec will drop the last commit, as we're pretty sure it's not really necessary.

**Comment by alecsloan** [comment]
force-push reflects the equals() and hashcode() commit being dropped

**Review by ethanbustad — APPROVED** [review]

**Pull request merged**